crates/bevy_reflect/src/impls/std.rs

```diff
@@ -166,7 +166,7 @@ impl<T: FromReflect> FromReflect for Vec<T> {
     }
 }
 
-impl<K: Reflect + Eq + Hash, V: Reflect> Map for HashMap<K, V> {
+impl<K: FromReflect + Eq + Hash, V: FromReflect> Map for HashMap<K, V> {
```

Comment:
I imagine this could be a breaking change. Maybe add a `Migration Guide` section on the PR comment explaining that key and value types now have a `FromReflect` bound instead of `Reflect`?